
  - name: test_aos_login_correct_login
    local_action:
      module: aos_login
      server: "{{ ansible_ssh_host }}"
      user: "{{ ansible_ssh_user }}"
      passwd: "{{ ansible_ssh_pass }}"
      port: 443
    register: test
    run_once: true

  - name: check_test_aos_login_correct
    assert:
      that:
        - test|success
        - test|changed
        - aos_session
        - test.aos_session

  - name: test_aos_login_incorrect_login
    local_action:
      module: aos_login
      server: "{{ ansible_ssh_host }}"
      user: "{{ ansible_ssh_user }}"
      passwd: "{{ ansible_ssh_pass_bad }}"
      port: 443
    register: test
    run_once: true
    ignore_errors: True

  - debug:
      msg: "{{ test.msg }}"

  - name: check_test_aos_login_incorrect
    assert:
      that:
        - test|failed
        - "'Invalid credentials' in test.msg"